dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/DataManager.kt

```diff
@@ -0,0 +1,131 @@
+package org.dataland.datalandbackend.service
+
+import org.dataland.datalandbackend.interfaces.DataManagerInterface
+import org.dataland.datalandbackend.interfaces.DataStoreInterface
+import org.dataland.datalandbackend.model.CompanyMetaInformation
+import org.dataland.datalandbackend.model.DataIdentifier
+import org.dataland.datalandbackend.model.DataSetMetaInformation
+import org.dataland.datalandbackend.model.StorableCompany
+import org.dataland.datalandbackend.model.StorableDataSet
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a data manager for Dataland including meta data storages
+ */
+@Component("DefaultManager")
+class DataManager(
+    @Autowired var dataStore: DataStoreInterface
+) : DataManagerInterface {
+    var dataMetaData = mutableMapOf<String, DataSetMetaInformation>()
+    var companyData = mutableMapOf<String, StorableCompany>()
+    private var companyCounter = 0
+
+    /*
+    ________________________________
+    Helper-methods:
+    ________________________________
+     */
+
+    private fun verifyCompanyIdExists(companyId: String): Boolean {
+        if (companyData.containsKey(companyId)) {
+            return true
+        }
+        return false
+    }
```

Comment:
warum nicht einfach return (companyData.containsKey(companyId)) bzw. warum braucht man das überhaupt das ist doch redundant?

Reply:
Das wird nur in Verbindung von diesem Block verwendet:
if (!verifyCompanyIdExists(companyId)) {
            throw IllegalArgumentException("The companyId: $companyId does not exist.")
        }
dann kann man doch das ganze als Funktion definieren und spart sich die redundanten if Blöcke

Reply:
Ich habe die Funktion jetzt auf `return (companyData.containsKey(companyId))` angepasst.

Reply:
Da würde ich thor innerhald der Funtion einpacken. Das passt zun Name der Methode und das wiederholt sich zusammen.

Reply:
ah ist schon gemacht. alles klar :)